controller/ResultController.php

```diff
@@ -0,0 +1,232 @@
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoLtiConsumer\controller;
+
+use common_exception_Error;
+use common_exception_MethodNotAllowed;
+use common_exception_NotFound;
+use common_user_auth_AuthFailedException;
+use oat\taoLti\models\classes\Lis\LisAuthAdapterFactory;
+use oat\taoLti\models\classes\Lis\LtiProviderUser;
+use oat\taoLtiConsumer\model\result\messages\LisOutcomeRequest;
+use oat\taoLtiConsumer\model\result\messages\LisOutcomeRequestParser;
+use oat\taoLtiConsumer\model\result\messages\LisOutcomeResponseInterface;
+use oat\taoLtiConsumer\model\result\operations\BasicResponse;
+use oat\taoLtiConsumer\model\result\operations\failure\BasicResponseSerializer;
+use oat\taoLtiConsumer\model\result\operations\OperationsCollection;
+use oat\taoLtiConsumer\model\result\ParsingException;
+use oat\taoLtiConsumer\model\result\ResultService as LtiResultService;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Request;
+use Slim\Http\StatusCode;
+use tao_actions_CommonModule;
+use tao_models_classes_UserException;
+use Throwable;
+use function GuzzleHttp\Psr7\stream_for;
+
+class ResultController extends tao_actions_CommonModule
+{
+    public const XML_CONTENT_TYPE = 'application/xml';
+
+    /**
+     * @noinspection PhpUnused
+     * @throws common_exception_MethodNotAllowed
+     * @throws tao_models_classes_UserException
+     */
+    public function manageResults()
+    {
+        if (!$this->isRequestPost()) {
+            throw new common_exception_MethodNotAllowed(null, 0, [Request::HTTP_POST]);
+        }
+
+        try {
+            $user = $this->authorizeUser($this->getPsrRequest());
+        } catch (tao_models_classes_UserException $userException) {
+            throw $userException;
+        } catch (Throwable $throwable) {
+            $this->response = $this->createInternalErrorResponse($throwable);
+            return;
+        }
+
+        $payload = (string) $this->getPsrRequest()->getBody();
+        $requestParser = $this->getRequestParser();
+
+        try {
+            $lisRequest = $requestParser->parse($payload);
+            $this->response = $this->processLisRequest(
+                $lisRequest,
+                // Require delivery execution to has the same tenant id (delivery property) as
+                // LtiProvider if specified for it
+                $user->getLtiProvider()->getTenantId()
```

Comment:
This extension should be not aware of any tenant related logic, please restructure code.

Reply:
Fixed, see https://github.com/oat-sa/extension-tao-deliver-connect/pull/37